spec/unit/application/master_spec.rb

```diff
@@ -314,11 +314,38 @@
         @master.main
       end
 
-      it "should drop privileges if running as root" do
-        Puppet.features.stubs(:root?).returns true
+      context "user privileges" do
+        before do
+          @user = Puppet::Type.type(:user).new(:name => 'puppet')
+          Puppet::Type.type(:user).expects(:new).returns @user
+        end
+
+        it "should drop privileges if running as root and the puppet user exists" do
+          Puppet.features.stubs(:root?).returns true
+          @user.expects(:exists?).returns true
+
+          Puppet::Util.expects(:chuser)
+
+          @master.main
+        end
+
+        it "should exit and log an error if running as root and the puppet user does not exist" do
+          Puppet.features.stubs(:root?).returns true
+          @user.expects(:exists?).returns false
 
+          expect { @master.main }.to raise_error(Puppet::Error, /Could not change user to puppet\. User does not exist and is required to continue\./)
+        end
+      end
+
+      it "should log a deprecation notice when running a WEBrick server" do
+        @user = Puppet::Type.type(:user).new(:name => 'puppet')
+        Puppet::Type.type(:user).expects(:new).returns @user
+        Puppet.features.stubs(:root?).returns true
+        @user.expects(:exists?).returns true
         Puppet::Util.expects(:chuser)
 
+        Puppet.expects(:deprecation_warning).with("The WEBrick Puppet Master is deprecated and will be removed in a future release.")
```

Comment:
@kylog I was wondering the same thing. If we do deprecate in code, would it be going out in 4.1.0?